code/datums/components/hallucinations.dm

```diff
@@ -0,0 +1,537 @@
+	buttcrab
+		fake_icon_state = "buttcrab"
+		should_attack = FALSE
+	bat
+		fake_icon_state = "bat"
+		get_name()
+			return pick("bat", "batty", "the roundest possible bat", "the giant bat that makes all of the rules")
+	snake
+		fake_icon_state = "rattlesnake"
+		get_name()
+			return pick("snek", "WHY DID IT HAVE TO BE SNAKES?!", "rattlesnake", "OH SHIT A SNAKE")
+
+	scorpion
+		fake_icon_state = "spacescorpion"
+		get_name()
+			return "space scorpion"
+
+	aberration
+		fake_icon_state = "aberration"
+		should_attack = FALSE
+		get_name()
+			return "transposed particle field"
+
+	capybara
+		fake_icon_state = "capybara"
+		should_attack = FALSE
+
+	frog
+		fake_icon_state = "frog"
+		should_attack = FALSE
+
+	disposing()
+		my_target = null
+		. = ..()
+
+/obj/fake_attacker/attackby()
+	step_away(src,my_target,2)
+	for(var/mob/M in oviewers(world.view,my_target))
+		boutput(M, "<span class='alert'><B>[my_target] flails around wildly.</B></span>")
+	my_target.show_message("<span class='alert'><B>[src] has been attacked by [my_target] </B></span>", 1) //Lazy.
+	return
+
+/obj/fake_attacker/Crossed(atom/movable/M)
+	..()
+	if (M == my_target)
+		step_away(src,my_target,2)
+		if (prob(30))
+			for(var/mob/O in oviewers(world.view , my_target))
+				boutput(O, "<span class='alert'><B>[my_target] stumbles around.</B></span>")
+
+
+/obj/fake_attacker/New(location, target)
+	..()
+	SPAWN(30 SECONDS)	qdel(src)
+	src.name = src.get_name()
+	src.my_target = target
+	if (src.fake_icon && src.fake_icon_state)
+		var/image/image = image(icon = src.fake_icon, loc = src, icon_state = src.fake_icon_state)
+		image.override = TRUE
+		target << image
+	step_away(src,my_target,2)
+	process()
+
+/obj/fake_attacker/proc/process()
+	if (!my_target)
+		qdel(src)
+		return
+	if (BOUNDS_DIST(src, my_target) > 0)
+		step_towards(src,my_target)
+	else
+		if (src.should_attack && prob(70) && !ON_COOLDOWN(src, "fake_attack_cooldown", 1 SECOND))
+			if (weapon_name)
+				if (narrator_mode)
+					my_target.playsound_local(my_target.loc, 'sound/vox/weapon.ogg', 40, 0)
+				else
+					my_target.playsound_local(my_target.loc, "sound/impact_sounds/Generic_Hit_[rand(1, 3)].ogg", 40, 1)
+				my_target.show_message("<span class='alert'><B>[my_target] has been attacked with [weapon_name] by [src.name] </B></span>", 1)
+				if (prob(20)) my_target.change_eye_blurry(3)
+				if (prob(33))
+					if (!locate(/obj/overlay) in my_target.loc)
+						fake_blood(my_target)
+			else
+				if (narrator_mode)
+					my_target.playsound_local(my_target.loc, 'sound/vox/hit.ogg', 40, 0)
+				else
+					my_target.playsound_local(my_target.loc, pick(sounds_punch), 40, 1)
+				my_target.show_message("<span class='alert'><B>[src.name] has punched [my_target]!</B></span>", 1)
+				if (prob(33))
+					if (!locate(/obj/overlay) in my_target.loc)
+						fake_blood(my_target)
+			attack_twitch(src)
+
+	if (src.should_attack && prob(10)) step_away(src,my_target,2)
+	SPAWN(0.3 SECONDS) .()
```

Comment:
```suggestion
	SPAWN(0.3 SECONDS)
		.()
```

Reply:
uhh, `.()` 👀

Reply:
I'm pretty sure the `.()` is intended as a gross way to a process loop. I'mma change it `src.process()` instead for clarity.

Reply:
```suggestion
	SPAWN(0.3 SECONDS)
		src.process()
```